test/syscalls/linux/proc.cc

```diff
@@ -511,6 +513,87 @@ TEST(ProcSelfAuxv, EntryValues) {
   EXPECT_EQ(i, proc_auxv.size());
 }
 
+
+// Just open and read a part of /proc/self/mem, check that we can read an item.
+TEST(ProcSelfMem, Basic) {
+  auto memfd =
+      ASSERT_NO_ERRNO_AND_VALUE(Open("/proc/self/mem", O_RDONLY));
+  char input[] = "hello-world";
+  char output[sizeof(input)];
+  ASSERT_THAT(pread(memfd.get(), output, sizeof(output), (off_t) (input)),
+              SyscallSucceedsWithValue(sizeof(input)));
+  ASSERT_STREQ(input, output);
+}
+
+// Perform read on /proc/[pid]/mem from same task group.
+TEST(ProcSelfMem, TaskGroup) {
+  int pfd1[2] = {};
+  int pfd2[2] = {};
+
+  ASSERT_THAT(pipe(pfd1), SyscallSucceeds());
+  ASSERT_THAT(pipe(pfd2), SyscallSucceeds());
+
+  // Create child process
+  pid_t const child_pid = fork();
+  if (child_pid == 0) {
+    // Close reading end of first pipe
+    close(pfd1[0]);
+
+    // Tell parent about location of input
+    char input[] = "hello-world";
+    off_t input_location = (off_t) input;
+    int res = WriteFd(pfd1[1], &input_location, sizeof(input_location));
+    EXPECT_EQ(res, sizeof(input_location)) << "bad size written to pipe fd";
+    EXPECT_THAT(close(pfd1[1]), SyscallSucceeds());
+
+    // Wait and close writing end of second pipe
+    wait(NULL);
+    EXPECT_THAT(close(pfd2[1]), SyscallSucceeds());
+
+    // Await parent OK to die
+    char ok = 0;
+    res = ReadFd(pfd2[0], &ok, sizeof(ok));
+    EXPECT_EQ(res, sizeof(ok)) << "bad size for ok signal";
+
+    // Close rest pipes
+    EXPECT_THAT(close(pfd2[0]), SyscallSucceeds());
+    _exit(0);
+  } else {
+    // In parent process.
+    ASSERT_THAT(child_pid, SyscallSucceeds());
+
+    // Close writing end of first pipe
+    EXPECT_THAT(close(pfd1[1]), SyscallSucceeds());
+
+    // Read target location from child
+    off_t target_location;
+    int res = ReadFd(pfd1[0], &target_location, sizeof(target_location));
+    EXPECT_EQ(res, sizeof(target_location)) << "bad size read from pipe fd";
+    // Close reading end of first pipe
+    EXPECT_THAT(close(pfd1[0]), SyscallSucceeds());
+
+    // Open /proc/pid/mem fd
+    std::string mempath = absl::StrCat("/proc/", child_pid, "/mem");
+    auto memfd =
+        ASSERT_NO_ERRNO_AND_VALUE(Open(mempath, O_RDONLY));
+    char output[12];
```

Comment:
Define a `char expected[] = "hello-world";` so that you can use `sizeof(expected)` here instead of a magic number, and also use `expected` below in the string comparison.